### DISCUSSION TITLE
Package issues with PIgments and atom-beautify

### DISCUSSION THREAD

**Discussion by SilentGlasses** [pulsar-edit/.github#121] [Community Package Help]
I've been having issues with two packages so far, both installed from instied Pulsare under the Install tab..

## atom-beautify

```
atom-beautify (1 deprecation)
Use customElements.define instead of document.registerElement see https://javascript.info/custom-elements

HTMLDocument.registerElement - file:///Applications/Pulsar.app/Contents/Resources/app.asar/static/index.js:99:12
registerElement - [/Users/framirez/.pulsar/packages/atom-beautify/node_modules/space-pen/lib/space-pen.js:66:88](https://github.com/Users/framirez/.pulsar/packages/atom-beautify/node_modules/space-pen/lib/space-pen.js:66:88)
Builder.openTag - [/Users/framirez/.pulsar/packages/atom-beautify/node_modules/space-pen/lib/space-pen.js:297:27](https://github.com/Users/framirez/.pulsar/packages/atom-beautify/node_modules/space-pen/lib/space-pen.js:297:27)
Builder.tag - [/Users/framirez/.pulsar/packages/atom-beautify/node_modules/space-pen/lib/space-pen.js:274:12](https://github.com/Users/framirez/.pulsar/packages/atom-beautify/node_modules/space-pen/lib/space-pen.js:274:12)
Function.div - [/Users/framirez/.pulsar/packages/atom-beautify/node_modules/space-pen/lib/space-pen.js:83:49](https://github.com/Users/framirez/.pulsar/packages/atom-beautify/node_modules/space-pen/lib/space-pen.js:83:49)
Function.content - [/Users/framirez/.pulsar/packages/atom-beautify/src/views/loading-view.coffee:19:19](https://github.com/Users/framirez/.pulsar/packages/atom-beautify/src/views/loading-view.coffee:19:19)
```

## Pigments

Doesn't render colors in CSS files..
![Screen Shot 21](https://user-images.githubusercontent.com/22822565/215618400-7bfdcbe5-5e53-4982-8265-2aed1cf6842a.png)

**Comment by confused-Techie** [comment]
Unfortunately there's only so much we can do when resolving errors in other packages.

Since those packages are made and maintained by someone else we don't control their code, and realistically with the updates happening on Pulsar some breakage is bound to happen.

I'd recommend opening issues on those package's repositories and see if the maintainers are willing to support the changes for Pulsar.

Otherwise if this is an actual bug in Pulsar we would be more than happy to have it found and patched, but based on the issue described and the error logs provided that doesn't seem likely.

**Reply by Daeraxa** [reply]
To add to the above, the best way of checking to see if it was any of our changes that broke it would be to see if the same package works on Atom 1.60 (not the very last 1.63 release which has an Electron bump independent of our Electron bump which also broke stuff).

**Reply by SilentGlasses** [reply]
It works in Atom for me...  Still does its thing in 1.63 as well with a deprecation error:

```
Use customElements.define instead of document.registerElement see https://javascript.info/custom-elements
```

Wondering now if that's the cause of it not working in Pulsar... I also opened a support request with the author [Support for Pulsar?](https://github.com/abe33/atom-pigments/issues/488)

### In Atom
![Screen Shot 23](https://user-images.githubusercontent.com/22822565/215812010-2e894d3f-a848-4a34-82c6-cd954fd159cb.png)

**Reply by confused-Techie** [reply]
Super interesting that this does work on Atom, especially considering it still works on 1.63, since most of the package breakage comes from Electron updates. So this may be an actual regression on our end rather than just a change from the update. So might be worth doing a little investigating into how this package works and what's causing it to break

**Comment by SilentGlasses** [comment]
I re-installed everything again last night in case there was something off on my end.  The atom-beautify issue went away, not sure what happened but, if it does not compute ... reboot seems to be true still. :P

**Reply by SilentGlasses** [reply]
In the interim, I switched to [atom-highlight-colors](https://github.com/brumm/atom-highlight-colors)

**Comment by confused-Techie** [comment]
So @SilentGlasses I've just wanted to test this a little bit, and interestingly enough, it seems that Pigments is working fine for me?

![image](https://user-images.githubusercontent.com/26921489/215918175-32b49723-8555-43c2-9829-dd7b712a6289.png)

So real fast to confirm what Pulsar Version and OS are you on?

As for me:

OS: Windows 10
Pulsar: 1.101.0-beta
Pigments: 0.40.6

**Reply by Spiker985** [reply]
Both Pigments and Atom-Beautify are standard on my machines, and have been working fine as well

OS: W11/Arch Linux
Version: 1.101.20230124/Built version as of 2023-01-29

**Reply by SilentGlasses** [reply]
I tried it out on my POP! OS laptop and it works as well with one deprecation warning.

**Comment by SilentGlasses** [comment]
OS: macOS 12.6.3 21G419 x86_64 
Kernel: 21.6.0 
Shell: zsh 5.8.1 
CPU: Intel i7-9750H (12) @ 2.60GHz 
GPU: Intel UHD Graphics 630, AMD Radeon Pro 5500M 
Memory: 18298MiB / 32768MiB

Pulsar ver: 1.101.0-beta

**Reply by confused-Techie** [reply]
Thanks! I'll try to find out if we have anyone with that specific version of MacOS that can investigate

**Comment by meadowsys** [comment]
seems to work for me! macOS 13.2 22D49 aarch64 (public beta) with Apple M1 Pro (the 10cpu/16gpu version)

**Comment by confused-Techie** [comment]
So @SilentGlasses it seems @Meadowsys with MacOS 13.2 is the closest we can find to test that specific version of MacOS.

Since none of us can seem to recreate the issue, is it possible you'd be able to provide the full logs from your dev tools window when using Pigments? Since you've provided logs for the other package which is helpful, this may be needed to track down what's actually broken.

It just seems super weird that it'd break only on that version of MacOS

**Comment by lemonjesus** [comment]
I have MacOS 12.6.2 and I am unable to reproduce this problem.

![image](https://user-images.githubusercontent.com/29158638/217385637-c83b6c13-c802-4eae-881f-f1f200eb7baa.png)

MacOS 12.6.2
Pulsar 1.101.0-beta x64
`atom-beautify` 0.33.4
`pigments` 0.40.6

I still get the "Use `customElements.define` instead of `document.registerElement`" deprecation warning for both `atom-beautify` and `pigments`, but it seems to work fine.

This is a fresh install from scratch on a computer that has never had Pulsar installed on it (I don't use Pulsar, I was just asked to try and replicate this by a friend since I have a relevant OS version). I also only have these two packages installed in the editor (beyond the stuff it comes with). So unless something significant changed between macOS 12.6.2 and 12.6.3, it's unlikely the OS version causing this.

**Reply by confused-Techie** [reply]
Thanks a ton for posting this, your help is appreciated!
So @SilentGlasses I'd say without further logs there may be nothing we can do to recreate this, unless you are able to determine exact steps that will replicate the issue.

Otherwise I'd recommend checking your Pulsar installation and any settings that may be having an effort on the editor.
